src/main.rs

```diff
@@ -195,13 +201,25 @@ fn start_ping_thread(
     })
 }
 
-fn get_host_ipaddr(host: &str) -> Result<String> {
+fn get_host_ipaddr(host: &str, force_ipv4: bool, force_ipv6: bool) -> Result<String> {
     let ipaddr: Vec<IpAddr> = match lookup_host(host) {
         Ok(ip) => ip,
         Err(_) => return Err(anyhow!("Could not resolve hostname {}", host)),
     };
-    let ipaddr = ipaddr.first();
-    Ok(ipaddr.unwrap().to_string())
+    let ipaddr = if force_ipv4 {
+        ipaddr
+            .iter()
+            .filter(|ip| matches!(ip, IpAddr::V4(_)))
+            .next()
+    } else if force_ipv6 {
+        ipaddr
+            .iter()
+            .filter(|ip| matches!(ip, IpAddr::V6(_)))
+            .next()
+    } else {
+        ipaddr.first()
+    };
+    Ok(ipaddr.expect("Could not resolve host to IP").to_string())
```

Comment:
I'd rather return an error here rather than using `expect()`. Couldn't we instead do:

```
match ipaddr {
    None => Err(anyhow!("could not resolve host to IP")),
    Some(addr) => Ok(addr)
}
```

Reply:
Absolutely, that makes much more sense. I was replacing the `unwrap()`s to have more signal and error is even better 👍